File: ui/src/app/models/Role.ts
import { Permission } from "./Permission";

export class Role {
  private id: string;
  private authority: string;
  private projectId: string;
  private permissions: Permission[];

  getId(): string {
    return this.id;
  }

  getProjectId(): string {
    return this.projectId;
  }

  getName(): string {
    return this.authority;
  }

  getPermissions(): Permission[] {
    return this.permissions;
  }

  setAuthority(authority: string) {
    this.authority = authority;
  }

  setProjectId(projectId: string) {
    this.projectId = projectId;
  }

  setPermissions(permissions: Permission[]) {
    console.log(this.permissions);
    this.permissions = permissions;
    console.log(this.permissions);
  }
}

File: back/auth/src/main/java/com/issuetracker/auth/model/Role.java
package com.issuetracker.auth.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;
import org.springframework.security.core.GrantedAuthority;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.Collection;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class Role implements GrantedAuthority, Serializable {
    private @NotBlank UUID id;
    private @NotBlank String authority;
    private @NotBlank UUID projectId;
    private @NotNull Collection<Permission> permissions;

    public Role(String authority, UUID projectId, Collection<Permission> permissions) {
        this.authority = authority;
        this.projectId = projectId;
        this.permissions = permissions;
    }

    public String getAuthority() {
        return authority;
    }
}
